be/src/exec/es/es_scroll_parser.cpp

```diff
@@ -504,6 +514,75 @@ Status ScrollParser::_append_bool_val(const rapidjson::Value& col, Column* colum
 
     return Status::OK();
 }
+
+Status ScrollParser::_append_array_val(const rapidjson::Value& col, const TypeDescriptor& type_desc, Column* column,
+                                       bool pure_doc_value) {
+    // Array type must have child type.
+    const auto& child_type = type_desc.children[0];
+    DCHECK(child_type.type != INVALID_TYPE);
+
+    // In ElasticSearch, n-dimensional array will be flattened into one-dimensional array.
+    // https://www.elastic.co/guide/en/elasticsearch/reference/8.3/array.html
+    // So we do not support user to create nested array column.
+    // TODO: We should prevent user to create nested array column in FE, but we don't do any schema validation now.
+    if (child_type.type == TYPE_ARRAY) {
+        std::string str = fmt::format("Invalid array format; Document slice is: {}.", json_value_to_string(col));
+        return Status::RuntimeError(str);
+    }
+
+    ArrayColumn* array = nullptr;
+
+    if (column->is_nullable()) {
+        auto* nullable_column = down_cast<NullableColumn*>(column);
+        auto* data_column = nullable_column->data_column().get();
+        NullData& null_data = nullable_column->null_column_data();
+        null_data.push_back(0);
+        array = down_cast<ArrayColumn*>(data_column);
+    } else {
+        array = down_cast<ArrayColumn*>(column);
+    }
+
+    auto* offsets = array->offsets_column().get();
+    auto* elements = array->elements_column().get();
+
+    if (pure_doc_value) {
+        RETURN_IF_ERROR(_append_doc_value_array_val(col, child_type, elements));
+    } else {
+        RETURN_IF_ERROR(_append_source_array_val(col, child_type, elements));
+    }
+
+    size_t new_size = elements->size();
+    offsets->append(new_size);
+    return Status::OK();
+}
+
+Status ScrollParser::_append_doc_value_array_val(const rapidjson::Value& col, const TypeDescriptor& child_type_desc,
+                                                 Column* column) {
+    for (auto& item : col.GetArray()) {
+        RETURN_IF_ERROR(_append_value_from_json_val(column, child_type_desc, item, true));
+    }
+    return Status::OK();
+}
+
+Status ScrollParser::_append_source_array_val(const rapidjson::Value& col, const TypeDescriptor& child_type_desc,
+                                              Column* column) {
+    if (col.IsNull()) {
```

Comment:
If col is null, is it treated as a null or an empty array?

Reply:
empty array, I have tested this situation in ut.

https://github.com/StarRocks/starrocks/blob/066d98ceaa748d183ed91ea56b89b19eb1c491ab/be/test/exec/es/es_scroll_parser_test.cpp#L148-L171

Reply:
I think we should return null if it doesn't exist in the ES

Reply:
We've already discussed it @RowenWoo .
We should keep the same behavior with doc_values.

Reply:
if the column does not exists in `_source` or `doc_value`, we return null, i think maybe this is the only one scenario

Reply:
"fields": { 
     //         "array": [] 
     //     } ,

this happen after version 7.x